cartographer_rviz/cartographer_rviz/submaps_display.h

```diff
@@ -32,6 +32,21 @@
 
 namespace cartographer_rviz {
 
+// This should be a private class in SubmapsDisplay, unfortunately, QT does not
+// allow for this.
+struct Trajectory : public QObject {
+  Q_OBJECT
+
+ public:
+  Trajectory(std::unique_ptr<::rviz::BoolProperty> property);
```

Comment:
@SirVer You can pass unique pointers by value in constructors? How does this exactly work?